### ISSUE TITLE
How would one resync the index.html after manually removing some files?

### ISSUE THREAD

**Issue by davidkim83** [pypa/bandersnatch#1109]
Hi, I'm trying to remove files I don't need to maintain; mainly filenames that contain the strings cp34, cp35, cp37, cp310.
They're essentially *.whl files that have nothing to do with the Python versions available in the underlying infrastructure.

I couldn't figure out how to filter these through the configuration files, so I created a shell script to remove them manually.
After this, I am faced with the problem of recreating the simple index.

Is there a way to do this -manual recreation of simple index- currently?
Or if you could give me an example of a filter configuration for removing such files, it'd be great.

**Comment by cooperlees** [comment]
Howdy, so this would cause you to dirty simple html files, but I think our plugins do to. But if you're infra doesn't need these binaries, pip in said infra should never select these wheels.

I feel the Regex filtering should do what you want:
- https://bandersnatch.readthedocs.io/en/latest/filtering_configuration.html#regex-filtering

```ini
[plugins]
enabled =
    regex_project

[filter_regex]
releases =
    .*cp3[14-7]0?.*
```

From my reading it excludes like you want ... Test away I feel. If it doesn't work how you need let us know and we can see if we can tune it.

**Comment by davidkim83** [comment]
Thanks for the prompt response!

Before your reply, I made modifications to the `filename_name.py` filter since it checks the filenames directly and makes partial matches to filter out packages.
It may not be the most elegant way to implement this, but its working for me so far.
I only got to write this reply now as my hands were full during the weekend.

Here's the code modification i made for future reference:

`filename_name.py`
```python

class ExcludePlatformFilter(FilterReleaseFilePlugin):
    """
    Filters releases based on regex patterns defined by the user.
    """

    name = "exclude_platform"

    _patterns: List[str] = []
    _packagetypes: List[str] = []

    # declare versions
    _python2xversions  = ['-cp2' , '-py2']
    _python27versions  = ['-cp27-' , '-py2.7']
    _python33versions  = ['-cp33-' , '-py3.3']
    _python34versions  = ['-cp34-' , '-py3.4']
    _python35versions  = ['-cp35-' , '-py3.5']
    _python36versions  = ['-cp36-' , '-py3.6']
    _python37versions  = ['-cp37-' , '-py3.7']
    _python38versions  = ['-cp38-' , '-py3.8']
    _python39versions  = ['-cp39-' , '-py3.9']
    _python310versions = ['-cp310-', '-py3.10']

...

    def initialize_plugin(self) -> None:
        """
        Initialize the plugin reading patterns from the config.
        """
...
            # exclude verions
            elif lplatform in ("2.x"):
                self._patterns.extend(self._python2xversions)

            elif lplatform in ("2.7"):
                self._patterns.extend(self._python27versions)

            elif lplatform in ("3.3"):
                self._patterns.extend(self._python33versions)

            elif lplatform in ("3.4"):
                self._patterns.extend(self._python34versions)

            elif lplatform in ("3.5"):
                self._patterns.extend(self._python35versions)

            elif lplatform in ("3.6"):
                self._patterns.extend(self._python36versions)

            elif lplatform in ("3.7"):
                self._patterns.extend(self._python37versions)

            elif lplatform in ("3.8"):
                self._patterns.extend(self._python38versions)

            elif lplatform in ("3.9"):
                self._patterns.extend(self._python39versions)

            elif lplatform in ("3.10"):
                self._patterns.extend(self._python310versions)
...
```

`bandersnatch.conf`
```yaml

[blocklist]
platforms =
    windows
    macos
    2.x
    3.3
    3.4
    3.5
    3.7
    3.9
    3.10

```

**Comment by cooperlees** [comment]
Awesome. Happy to accept this if you're willing to do some unit tests + documentation.

**Comment by davidkim83** [comment]
Hi, its my first time contributing anything to GitHub, but I went away and made a [PR](https://github.com/pypa/bandersnatch/pull/1110).
Is there anything else I should provide?

**Comment by davidkim83** [comment]
Will be resolved with new release including #1110

**Comment by wudisheng** [comment]
FYI: Filtering python version patterns in filenames _might_ be too aggressive in certain cases.

For example, `murmurhash2` has a filename like `...-cp36-abi3-...` which supports Python 3.8, 3.10, etc., but it will be filtered out if I put `py3.6` in the blocklist (resulting in a build-from-source action that failed due to missing toolchains).

IIUC, the _correct_ purpose of filtering out some python versions is to eliminate support of some (lower) versions, but in the case above, higher versions are affected negatively.

Unfortunately, I do not have a clear workaround suggestion at this time. Hope that someone with good knowledge on the naming of Python packages could help.
